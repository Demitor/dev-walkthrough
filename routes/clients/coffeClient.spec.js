const {
    fetchCoffeeByCountry,
    fetchCoffeeFilteredByCountry,
    fetchCoffeeSortedByRating,
    fetchAllCoffeeTypes
} = require('./coffeeClient');

describe('coffeeClient', () => {
    it('should get coffee types by country', () => {
        const result = fetchCoffeeByCountry();
        expect(result).toMatchSnapshot();
    });

    it('should get coffee types filtered by country', () => {
        const result = fetchCoffeeFilteredByCountry('FI');
        expect(result).toMatchSnapshot();
    });

    it('should get coffee types sorted by rating', () => {
        const result = fetchCoffeeSortedByRating();
        expect(result).toMatchSnapshot();
    });

    it('should get all coffee types', () => {
        const result = fetchAllCoffeeTypes();
        expect(result).toMatchSnapshot();
    });
});