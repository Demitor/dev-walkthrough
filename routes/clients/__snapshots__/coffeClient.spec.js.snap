// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`coffeeClient should get all coffee types 1`] = `
Array [
  Object {
    "country": "FI",
    "description": "This coffee bean with low caffeine and a smoother taste is aromatic and delicious. 80% of the coffee in the world is produced from these types of beans.",
    "rating": 4,
    "type": "arabica",
  },
  Object {
    "country": "FI",
    "description": "This type of coffee, which contains 2.5% more caffeine than other types, has a pretty strong taste.",
    "rating": 3.2,
    "type": "robusta",
  },
  Object {
    "country": "SE",
    "description": "Excelsa grows mostly in Southeast Asia and accounts for a mere 7% of the world's coffee circulation. It is largely used in blends in order to give the coffee an extra boost of flavor and complexity, better affecting the middle and back palate.",
    "rating": 1.9,
    "type": "excelsa",
  },
  Object {
    "country": "SP",
    "description": "Liberica is a low yield type of coffee compared to Arabica and Robusta.",
    "rating": 1.8,
    "type": "liberica",
  },
]
`;

exports[`coffeeClient should get coffee types by country 1`] = `
Object {
  "FI": Array [
    Object {
      "country": "FI",
      "description": "This coffee bean with low caffeine and a smoother taste is aromatic and delicious. 80% of the coffee in the world is produced from these types of beans.",
      "rating": 4,
      "type": "arabica",
    },
    Object {
      "country": "FI",
      "description": "This type of coffee, which contains 2.5% more caffeine than other types, has a pretty strong taste.",
      "rating": 3.2,
      "type": "robusta",
    },
  ],
  "SE": Array [
    Object {
      "country": "SE",
      "description": "Excelsa grows mostly in Southeast Asia and accounts for a mere 7% of the world's coffee circulation. It is largely used in blends in order to give the coffee an extra boost of flavor and complexity, better affecting the middle and back palate.",
      "rating": 1.9,
      "type": "excelsa",
    },
  ],
  "SP": Array [
    Object {
      "country": "SP",
      "description": "Liberica is a low yield type of coffee compared to Arabica and Robusta.",
      "rating": 1.8,
      "type": "liberica",
    },
  ],
}
`;

exports[`coffeeClient should get coffee types filtered by country 1`] = `
Array [
  Object {
    "country": "FI",
    "description": "This coffee bean with low caffeine and a smoother taste is aromatic and delicious. 80% of the coffee in the world is produced from these types of beans.",
    "rating": 4,
    "type": "arabica",
  },
  Object {
    "country": "FI",
    "description": "This type of coffee, which contains 2.5% more caffeine than other types, has a pretty strong taste.",
    "rating": 3.2,
    "type": "robusta",
  },
]
`;

exports[`coffeeClient should get coffee types sorted by rating 1`] = `
Array [
  Object {
    "country": "FI",
    "description": "This coffee bean with low caffeine and a smoother taste is aromatic and delicious. 80% of the coffee in the world is produced from these types of beans.",
    "rating": 4,
    "type": "arabica",
  },
  Object {
    "country": "FI",
    "description": "This type of coffee, which contains 2.5% more caffeine than other types, has a pretty strong taste.",
    "rating": 3.2,
    "type": "robusta",
  },
  Object {
    "country": "SE",
    "description": "Excelsa grows mostly in Southeast Asia and accounts for a mere 7% of the world's coffee circulation. It is largely used in blends in order to give the coffee an extra boost of flavor and complexity, better affecting the middle and back palate.",
    "rating": 1.9,
    "type": "excelsa",
  },
  Object {
    "country": "SP",
    "description": "Liberica is a low yield type of coffee compared to Arabica and Robusta.",
    "rating": 1.8,
    "type": "liberica",
  },
]
`;
